### DISCUSSION TITLE
Transcoding HDR videos (iPhone 12/13)

### DISCUSSION THREAD

**Discussion by kzaoaai** [photoprism/photoprism#1765] [Q&A] [answered]
Hello,

Anyone having color issues when transcoding videos shot on an iPhone with HDR turned on? The colors are off and very bright.

**Comment by lastzero** [comment, marked as answer]
Sounds plausible since Apple now uses the P3 color space by default:

- https://en.wikipedia.org/wiki/DCI-P3

When transcoding, it needs to be preserved or converted in order to keep the original colors. We currently don't have time to look into this in detail as we focus on multi user support for the rest of the year. You are most welcome to perform research and tests, so that we ideally only need to change a few ffmpeg parameters to make it work for you.

These links may help:

- https://www.google.com/search?q=P3+apple+color
- https://www.google.com/search?q=P3+color+space
- https://trac.ffmpeg.org/wiki/colorspace
- https://stackoverflow.com/questions/61834623/how-to-use-ffmpeg-colorspace-options

Alternatively, you can use an Apple certified transcoder to manually create AVC files using the right color space.

**Reply by lastzero** [reply]
related issue with images / thumbnails: https://github.com/photoprism/photoprism/issues/1474

**Reply by kzaoaai** [reply]
This out of my depth, but I'll look into this when time allows. I appreciate the direction and taking the time to respond.
